pennylane/templates/repeat.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``repeat`` template constructor.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.templates.decorator import template
+
+###################
+
+
+@template
+def repeat(circuit, depth, *args, **kwargs):
+    r"""Repeatedly applies a circuit containing quantum gates or templates.
+
+    Args:
+        circuit (function): A function that applies the quantum gates/templates being repeated.
+        depth (int): The number of times the circuit is repeatedly applied.
+        *args: Dynamic parameters that are passed into ``circuit`` each time it is
+               repeated (see UsageDetails for more information). Each dynamic argument
+               must be a list of first dimension equal to ``depth``, with the :math:`j`-th element of the list
+               corresponding to the value of the argument the :math:`j`-th time the circuit
+               is applied.
+        **kwargs: Static parameters that are passed into ``circuit`` each time it is
+                  repeated (see UsageDetails for more information).
+
+    Raises:
+        ValueError: if inputs do not have the correct format
+
+    .. UsageDetails::
+
+        **Repeating Gates**
+
+        To repeatedly apply a collection of gates/templates, a function applying each of these
+        operations must first be defined. For example, we can define the following circuit:
+
+        .. code-block:: python3
+
+            import pennylane as qml
+            import numpy as np
+
```

Comment:
Maybe I'm just so used to it, but this looks like a faulty quantum function again :)

Maybe one could name it 'gate_sequence', or 'subroutine'?

Reply:
Or `ansatz`!! You use that below and it is very suggestive of the main use case!